examples/vision/edsr.py

```diff
@@ -0,0 +1,320 @@
+"""
+Title: Enhanced Deep Residual Networks for Single Image Super-Resolution
+Author: Gitesh Chawda
+Date created: 16-03-2022
+Last modified: 16-03-2022
+Description: Implementing EDSR model on DIV2K Dataset.
+"""
+
+"""
+## Introduction 
+
+Enhanced Deep Residual Networks for Single Image Super-Resolution
+[EDSR](https://arxiv.org/abs/1707.02921) by Bee Lim, Sanghyun Son, Heewon Kim, Seungjun
+Nah, Kyoung Mu Lee.
+
+The EDSR architecture is based on the SRResNet architecture, consisting of multiple
+residual blocks. It doesn't use batch normalization layers (input and output have similar
+distributions, normalizing intermediate features may not be desirable) instead it uses
+constant scaling layers to ensure stable training. An L1 loss function (absolute error)
+is used instead of L2 (MSE), the authors showed better performance empirically and it
+requires less computation.
+
+In this code example we will implementing base model that includes just 16 ResBlocks and
+64 channels.
+
+Alernatively, you can also build a ESPCN Model as shown in the Keras example [Image
+Super-Resolution using an Efficient Sub-Pixel CNN
+](https://keras.io/examples/vision/super_resolution_sub_pixel/#image-superresolution-using
+-an-efficient-subpixel-cnn). As per the survey paper EDSR performs well than ESPCN. Paper
+: 
+
+[A comprehensive review of deep learningbased single image
+super-resolution](https://arxiv.org/abs/2102.09351)
+
+Camparison Graph :
+
+<img src="https://dfzljdn9uc3pi.cloudfront.net/2021/cs-621/1/fig-11-2x.jpg" width="500" />
+"""
+
+"""
+## Imports
+"""
+
+import numpy as np
+import tensorflow as tf
+import tensorflow_datasets as tfds
+import matplotlib.pyplot as plt
+
+from tensorflow import keras
+from tensorflow.keras import layers
+from tensorflow.keras.models import Model
+
+AUTOTUNE = tf.data.AUTOTUNE
+
+"""
+## Downloading Training Dataset
+
+Using DIV2K Dataset which is a popular single-image super-resolution dataset which
+contains 1,000 images with different scenes and is splitted to 800 for training, 100 for
+validation and 100 for testing. This dataset contains low resolution images with
```

Comment:
The text copy has some issues -- would you be able to get help from someone to improve it?

Reply:
Yeah